test/controllers/songs_controller_test.rb

```diff
@@ -72,6 +72,16 @@ class SongsControllerTest < ActionController::TestCase
     assert_not_nil flash[:success]
   end
 
+  test "editing a song should result in the song in the database with a different name" do
+    post_new_song_form
+    song = Song.find_by_name("New Song Just Posted")
```

Comment:
You can just find any old song already in the database with `songs(:song_name)` and do the update on it instead of posting it first. I created `post_new_song_form` as a helper for testing the create endpoint only, not as a helper to put data in the database as that is what the song fixtures are for.